### ISSUE TITLE
Configure script: cross-compile with external ACE+TAO

### ISSUE THREAD

**Issue by adgnaf** [objectcomputing/OpenDDS#1288]
Cross compilation for ARM linux is failed due to `opendds_idl` is broken. 
The symbolic link is ${DDS_ROOT}/build/host/bin/opendds_idl which is linked to ../../target/dds/opendds_idl, But there is no executable named "opendds_idl" in "target/dds/". 

The configure command executed as follows, 
```shell
./configure  --ace=${ACE_ROOT} --tao=${ACE_ROOT}/TAO  --mpc=${ACE_ROOT}/MPC --target=linux-cross --target-compiler=/usr/bin/arm-linux-gnueabihf-g++ --prefix=/some/path
```

ACE/TAO for host and target(arm) is both compiled and worked.

**Comment by adgnaf** [comment]
Found a workaround. 

After manually linked ${DDS_ROOT}/build/host/bin/opendds_idl to `opendds_idl` from previous host build, the compilation passed.

**Comment by adgnaf** [comment]
But this still seems to be an issue.

**Comment by mitza-oci** [comment]
> Cross compilation for ARM linux is failed due to `opendds_idl` is broken.
> The symbolic link is ${DDS_ROOT}/build/host/bin/opendds_idl which is linked to ../../target/dds/opendds_idl, But there is no executable named "opendds_idl" in "target/dds/".
> 
> The configure command executed as follows,
> 
> ```shell
> ./configure  --ace=${ACE_ROOT} --tao=${ACE_ROOT}/TAO  --mpc=${ACE_ROOT}/MPC --target=linux-cross --target-compiler=/usr/bin/arm-linux-gnueabihf-g++ --prefix=/some/path
> ```
> 
> ACE/TAO for host and target(arm) is both compiled and worked.

Does that mean you have compiled ACE+TAO before configuring OpenDDS?  I'm not sure that will work with cross-compilation, but we can try it.

Is this using master or a release?

**Comment by adgnaf** [comment]
> 
> 
> > Cross compilation for ARM linux is failed due to `opendds_idl` is broken.
> > The symbolic link is ${DDS_ROOT}/build/host/bin/opendds_idl which is linked to ../../target/dds/opendds_idl, But there is no executable named "opendds_idl" in "target/dds/".
> > The configure command executed as follows,
> > ```shell
> > ./configure  --ace=${ACE_ROOT} --tao=${ACE_ROOT}/TAO  --mpc=${ACE_ROOT}/MPC --target=linux-cross --target-compiler=/usr/bin/arm-linux-gnueabihf-g++ --prefix=/some/path
> > ```
> > 
> > 
> > ACE/TAO for host and target(arm) is both compiled and worked.
> 
> Does that mean you have compiled ACE+TAO before configuring OpenDDS? I'm not sure that will work with cross-compilation, but we can try it.
> 
> Is this using master or a release?

Yes. I have compiled ACE+TAO for ARM beforehand.

**Comment by mitza-oci** [comment]
I could see adding support for this.  Would it need an additional `--host-ace=` argument or assume that `--ace=` has a `build/host` and `build/target`?  If we can agree on requirements is it something you would like to contribute?

**Comment by rspydir** [comment]
I had a similar problem. ACE+TAO and opendds is part of our project's dependencies. We to produce native and ARM targets. Building ACE+TAO native first leaves artifacts that result the above error when trying to cross compile opendds for ARM. Our solution was to clone the ACE+TAO source tree in the opendds folder, then run a script that deletes build artifacts from the clone. We diff'ed the source distribution with the post-build state to find what files need to be deleted and packaged a bunch of 'rm -rf' statements in a clean_acetao.sh script, which is run as part of the opendds cross compile build process. It's not pretty, but it works consistently.

**Comment by mitza-oci** [comment]
> 
> 
> I had a similar problem. ACE+TAO and opendds is part of our project's dependencies. We to produce native and ARM targets. Building ACE+TAO native first leaves artifacts that result the above error when trying to cross compile opendds for ARM. Our solution was to clone the ACE+TAO source tree in the opendds folder, then run a script that deletes build artifacts from the clone. We diff'ed the source distribution with the post-build state to find what files need to be deleted and packaged a bunch of 'rm -rf' statements in a clean_acetao.sh script, which is run as part of the opendds cross compile build process. It's not pretty, but it works consistently.

Have you considered using git to manage which files are added and clean them (`git clean ...`)?

**Comment by iguessthislldo** [comment]
I added a `--host-ace` option to the configure script in #1446, which is in OpenDDS 3.14.

**Comment by mitza-oci** [comment]
Closed, fixed in #1446 

@iguessthislldo please reopen if this isn't right
